python/tank/descriptor/descriptor.py

```diff
@@ -371,6 +376,40 @@ def find_latest_version(self, constraint_pattern=None):
         latest._io_descriptor = self._io_descriptor.get_latest_version(constraint_pattern)
         return latest
 
+    def find_latest_cached_version(self, constraint_pattern=None):
```

Comment:
I am not entirely sure - one difference is how the return value is handled - in the case of a workflow around `find_latest_version()` you can assume that you always get a Descriptor object back, or an exception is raised to indicate - typically because you cannot connect to determine the version. But the expectation is that there is always a latest version available, unless something is wrong (cannot reach remote system or your descriptor uri is invalid etc).

For `find_latest_cached_version()` on the other hand, it's not strange that there isn't a version available and in this case you'll get None back.

So I think code would typically look like this:

```python
descriptor = create_descriptor(uri)

if descriptor.has_remote():
   # get latest from server
   latest = descriptor.find_latest_version()
else:
  # no remote connection
  latest = descriptor. find_latest_cached_version()
  if latest is None:
    raise Exception("No versions cached")
```

This is basically the logic that the `resolve_latest` flag implements now. Do you think that makes sense?